Shared/DatabaseDefintion/EntityModel/Repositories/CartRepository.cs

```diff
@@ -0,0 +1,113 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Threading;
+using System.Threading.Tasks;
+using DatabaseDefinition.EntityModel.Repositories.Interfaces;
+using DatabaseDefintion.EntityModel.Database;
+using Microsoft.EntityFrameworkCore;
+using TM = TransferModel;
+
+namespace DatabaseDefinition.EntityModel.Repositories;
+
+public class CartRepository : ICartRepository
+{
+    private readonly AppProjectDbContext dbContext;
+
+    public CartRepository(AppProjectDbContext dbContext)
+    {
+        this.dbContext = dbContext ?? throw new ArgumentNullException(nameof(dbContext));
+    }
+
+    public async Task AddItemToCartAsync(string userId, long itemId, CancellationToken cancellationToken)
+    {
+        _ = await dbContext.Items.Select(ItemMappings.MapItem)
+                                .SingleOrDefaultAsync(x => x.Id == itemId, cancellationToken)
+                                .ConfigureAwait(false) ??
+                                    throw new ArgumentException($"Item with id '{itemId}' does not exist.");
+        _ = await dbContext.Users.SingleOrDefaultAsync(x => x.Id == userId, cancellationToken)
+                                .ConfigureAwait(false) ??
+                                    throw new ArgumentException($"User with id '{userId}' does not exist.");
+        var cartItemLink = await dbContext.CartItemLinks.SingleOrDefaultAsync(x => x.ItemId == itemId && x.UserId == userId, cancellationToken)
+                                                        .ConfigureAwait(false);
+        if (cartItemLink != null)
+            cartItemLink.Amount++;
+        else
+            await dbContext.CartItemLinks.AddAsync(new CartItemLink() { UserId = userId, ItemId = itemId, Amount = 1 });
+        await dbContext.SaveChangesAsync(cancellationToken).ConfigureAwait(false);
+    }
+
+    public async Task<IEnumerable<TM.CartItemLink>> GetCartItemLinksAsync(string userId, CancellationToken cancellationToken)
+    {
+        _ = await dbContext.Users.SingleOrDefaultAsync(x => x.Id == userId, cancellationToken)
+                                .ConfigureAwait(false) ??
+                                    throw new ArgumentException($"User with id '{userId}' does not exist.");
+        var cartItemLinks = await dbContext.CartItemLinks.Select(CartItemLinkMappings.MapCartItemLink)
+            .Where(x => x.UserId == userId)
+            .ToArrayAsync(cancellationToken)
+            .ConfigureAwait(false);
+        foreach (var cartItemLink in cartItemLinks)
+            cartItemLink.Item = await dbContext.Items.Select(ItemMappings.MapItem)
```

Comment:
Kann man mit einem .Include lösen. Aber ich bin mir über die reichte Umsetzung mit dem Übertragen von item in TM.Item auch noch nicht im klaren. Sollten wir auf jeden Fall mal die Köpfe zusammenstecken und eine gute Lösung finden. Ich werde das Problem beim Platfrom Dingen auch haben.